typing/parmatch.ml

```diff
@@ -1007,8 +1007,8 @@ let build_other ext env =
             0 succ d env
       | Constant Const_int32 _ ->
           build_other_constant
-            (function Constant(Const_int32 i) -> i | _ -> assert false)
-            (function i -> Tpat_constant(Const_int32 i))
+            (function Constant(Const_int32 (i, _)) -> i | _ -> assert false)
+            (function i -> Tpat_constant(Const_int32 (i, "<TODO: parmatch.ml>")))
```

Comment:
TODO: Figure out what needs to be done here.

Reply:
This code is part of the exhaustivity-checking logic; `build_other` is called when we know that a given list of pattern (`env`) does not cover all cases, and it has to produce a counter-example, a pattern that is disjoint from all patterns in `env`. In the case where the patterns in `env` are constant int32 pattern, we find a counter-example that is also a constant int32 pattern, by looking at all numbers from `0l` in turn, until we find a constant pattern that is disjoint from the patterns in `env`.

In this case I think that the more reasonable thing to do is to keep a standard numeric representation for those made-up counter-example, instead of trying to respect the style of the input patterns in `env` (you can get their string representation, the first one for example is in the wildcard of the `Const_int32 _` pattern a few lines ago; but the patterns in `env` may use different styles, and trying to reverse-engineer a style and reuse it for made-up numbers sounds like more work than it is worth).

Reply:
Thank you for the explanation! I agree with your conclusion.